### ISSUE TITLE
Accessible issue with icon block link

### ISSUE THREAD

**Issue by NicktheGeek** [nk-crew/ghostkit#152]
The icon block can be set to have a link, which is great! However, the link does not have discernible text for screen readers and no way to set this. 
![Screen Shot 2022-08-24 at 9 19 31 AM](https://user-images.githubusercontent.com/835568/186429383-e46d8f15-4a51-4443-a8bc-77c0f5c26ffd.png)
![Screen Shot 2022-08-24 at 9 21 49 AM](https://user-images.githubusercontent.com/835568/186429388-f065cd02-7993-40a5-94fc-9f36b9b2b2f5.png)

This needs a new field added for the "link label" that can either be inserted in using screen reader text or added via an aria-label. 

This will enable screen reader users to know what the link is for. Otherwise it reads out the full URL, which isn't super helpful.

https://dequeuniversity.com/rules/axe/3.5/link-name
https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html
